Updated tests and backend route

sidsun1 opened the pull request:
## Summary
- Implemented more rigorous checks for POST requests being made to the flask server including whitespace, null, and incorrect type early returns
- Updated curl shell script to utilize jq and read from keys within a dummy JSON payload object to post realistic-looking (malicious) emails for the model to utilize as additional tests [NOTE: The model currently treats these very scam-likely emails as only slightly tending to be scam likely, hovering around ~0.6-0.7 probability, and even going as low as 0.46 in one case]
- Added an `INSTRUCTIONS.md` file to run tests easier locally, combining shell scripts with a Python-based suite of tests
- Updated test suite with Pytest, used pytest.fixture and pytest.parametrize to reduce duplicate testing code and prevent unnecessary backend API calls

## Related Issues/PRs
- #7 
- #13

sidsun1 commented:
Merging should be ready to go for now, the `/predict` API route has basically been fully tested. The only other significant consideration would be to add a classification key to be sent over to the frontend, but that requires simple numerical comparisons which could be dealt with there. cURL tests could be improved by running them with the error cases, however, the Pytest testcases so far accomplish this.

shreyasun reviewed (approved):
Just lint but looks great ✨

sidsun1 commented:
Added ruff to the environment for linting and fixed module formatting, run `pip install -r requirements.txt` when in your virtual environment and `ruff check` to run this test.
